### PR TITLE
Rework APIs to be sure to include everyone in the course tutor leaderboard

### PR DESCRIPTION
**Pull request by rpadovani**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: the project builds without code style warnings.
- [x] ~I updated the documentation and models.~
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~I added (end-to-end) test cases for the new functionality.~

### Motivation and Context
The course leaderboard did not include every tutor, now it does

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java [hunk 1]
```diff
@@ -306,7 +306,7 @@ private void addLatestResultToParticipation(Participation participation) {
         if (!courseService.userHasAtLeastTAPermissions(course)) {
             throw new AccessForbiddenException("You are not allowed to access this resource");
         }
-        List<Participation> participations = participationService.findByCourseIdWithRelevantResults(courseId);
+        List<Participation> participations = participationService.findByCourseIdWithRelevantResults(courseId, false);
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java]: This is a very performance heavy operation. In a course with 2000 participants and 80 exercises, this loads a lot of data. And now it gets slower. Can you please split the 2 cases, because here we don't need the assessor?

File: src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java [hunk 2]
```diff
@@ -509,10 +509,9 @@ public Participation findOneByExerciseIdAndStudentLogin(Long exerciseId, String
                             // we are only interested in results with completion date and with score
                             continue;
                         }
-                        if (participation.getExercise() instanceof QuizExercise) {
-                            // in quizzes we take all rated results, because we only have one! (independent of later checks)
-                        }
-                        else if (participation.getExercise().getDueDate() != null) {
+
+                        // in quizzes we take all rated results, because we only have one! (independent of later checks)
+                        if (!(participation.getExercise() instanceof QuizExercise) && participation.getExercise().getDueDate() != null) {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java]: please revert this change, we intentionally wrote this code like this

File: src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java (hunk 1 above)
Comment (rpadovani) [reply src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java]: Done

File: src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java (hunk 2 above)
Comment (rpadovani) [reply src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java]: Done

**Comment by rpadovani** [comment]
@krusche I addressed your comments
